internal/application/controlplane_service.go

```
@@ -0,0 +1,493 @@
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if request.Version != 0 && request.Version != controlplane.SchemaVersion {
+		return controlplane.ActionResult{}, fmt.Errorf("control plane: unsupported request schema version %d", request.Version)
+	}
+	request.Version = controlplane.SchemaVersion
+	objective := strings.TrimSpace(request.Arguments["objective"])
+	if (request.ActionID == "route" || request.ActionID == "run") && objective == "" {
+		return controlplane.ActionResult{}, errors.New("control plane: objective is required")
+	}
+	requestCtx := ctx
```

golangci-lint (2.13.2)

[error] ineffectual assignment to requestCtx

(ineffassign)